frontend/src/components/advanced-auto-filter/brand-details/brand-details.tsx

```diff
@@ -1,90 +1,87 @@
-import React, { FC, useEffect, useState } from 'react';
+import React, { FC } from 'react';
 
 import { AutocompleteValueType } from '@common/types/cars/autocomplete.type';
+import { BrandDetailsType } from '@common/types/cars/brand-details.type';
 import { AutocompleteInput } from '@components/common/autocomplete-input/autocomplete-input';
+import { SelectField } from '@components/common/select-field/select-field';
+import { getValueById } from '@helpers/get-value-by-id';
 import {
   useGetBrandsQuery,
   useGetModelsOfBrandQuery,
 } from '@store/queries/cars';
 
 type Props = {
-  onBrandDetailsChange: (data: {
-    id: string;
-    brandId: string;
-    modelId: string;
-  }) => void;
+  onBrandDetailsChange: (data: BrandDetailsType) => void;
   id: string;
+  selectedBrandId: string;
+  selectedModelId: string;
 };
 
-const BrandDetails: FC<Props> = ({ onBrandDetailsChange, id }) => {
-  const [selectedModelId, setSelectedModelId] = useState('');
-  const [selectedBrandId, setSelectedBrandId] = useState('');
-
+const BrandDetails: FC<Props> = ({
+  onBrandDetailsChange,
+  id,
+  selectedBrandId,
+  selectedModelId,
+}) => {
   const { data: brands, isLoading } = useGetBrandsQuery();
   const { data: models } = useGetModelsOfBrandQuery(selectedBrandId, {
     skip: !selectedBrandId,
   });
 
+  const selectedBrandName = getValueById(brands || [], selectedBrandId);
+  const selectedModelName = getValueById(models || [], selectedModelId);
+
   const handleSelectBrand = (data: AutocompleteValueType): void => {
-    setSelectedBrandId(data?.id || '');
-    setSelectedModelId('');
+    onBrandDetailsChange({
+      id,
+      brandId: data?.id || '',
+      modelId: '',
+    });
   };
 
   const handleSelectModel = (data: AutocompleteValueType): void => {
-    setSelectedModelId(data?.id || '');
-  };
-
-  useEffect(() => {
     onBrandDetailsChange({
       id,
       brandId: selectedBrandId,
-      modelId: selectedModelId,
+      modelId: data?.id || '',
     });
-  }, [selectedModelId, selectedBrandId]);
-
-  if (isLoading) return <h1>Loading...</h1>;
-
-  const selectedBrandName = (): AutocompleteValueType => {
-    const brand = brands?.find((brand) => brand.id === selectedBrandId);
-
-    return {
-      label: brand?.name || '',
-      id: brand?.id || '',
-    };
   };
 
-  const selectedModelName = (): AutocompleteValueType => {
-    const model = models?.find((model) => model.id === selectedModelId);
-
-    return {
-      label: model?.name || '',
-      id: model?.id || '',
-    };
-  };
+  if (isLoading) return <h1>Loading...</h1>;
```

Comment:
Why we don't use `<CircularProgress />` from mui instead of plain `Loading...` ?

Reply:
It was just temporary, I'll change it now